geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/FilterHelper.scala

```diff
@@ -318,5 +320,27 @@ object FilterHelper {
       case f: Filter => Seq(f)
     }
   }
+
+  def tryMergeGeometryFilters(filts: Seq[Filter]): Seq[Filter] = {
+    import org.geotools.data.DataUtilities._
+    import scala.collection.JavaConversions._
+    val filtFactory = CommonFactoryFinder.getFilterFactory2
+
+    def getAttrName(l: BBOX): String = propertyNames(l).head.getPropertyName
+
+    filts match {
+      // if we have two bbox filters as is common in WMS queries, merge them by intersecting the bounds
+      case Seq(l: BBOX, r: BBOX) if getAttrName(l) == getAttrName(r) =>
```

Comment:
If Intersects is used, will that result in a filter.primary having a Seq(BBOX, BBOX)?  This is intentionally limited.  I feel like there's gotta be code down in geotools that solves this for us.  However, I looked at the FilterToSQL and it looks like it punts to the database.